### DISCUSSION TITLE
Incremental direct message backups

### DISCUSSION THREAD

**Discussion by erincerys** [Tyrrrz/DiscordChatExporter#912] [General]
This post is to talk about incremental backups for message exports. I'll talk about why I'm thinking about, make an argument for it being a valuable feature, and also elaborate on there being historical support for it.

My account is over 4 years old, and a full export is quite time consuming. The earliest DM channel that the program always exports first is with an ex who I was with for many years. I'm presently running an export as I type this. It's been over 15 minutes, and it's only 17% finished.

Even when reusing pre-existing media that has been exported in a prior run (`exportdm --media --reuse-media`), the app appears not to scan the directory to do an incremental export of messages. It starts over from the beginning, collecting all messages in all channels. This is useful for catching edits, but quite inefficient when that isn't desired given the use case (e.g. see #911)

Incremental backups have been suggested in the past (#564) and support for them was refused, citing a workaround being the `--after` argument. That's a valid workaround and can be worked into scripting of guild exports. It isn't available for the `exportdm` command, though.

There is a case for supporting this in that the sensing and reuse of media is already supported for this exact same goal - efficiency (#395). As I understand it though, Tyrrrz had their hands full with other life circumstances even before war broke out, so it'll likely fall on contributors to take the reigns if they'd like this to happen.

**Comment by 96-LB** [comment]
> Incremental backups have been suggested in the past (#564) and support for them was refused, citing a workaround being the `--after` argument. That's a valid workaround and can be worked into scripting of guild exports. It isn't available for the `exportdm` command, though.

Are you sure the `--after` flag isn't available with `exportdm`? I believe it's intended to be.

**Reply by erincerys** [reply]
You're right... it indeed does. Not sure how I missed that.
